### DISCUSSION TITLE
Running an experiment offline on a new machine

### DISCUSSION THREAD

**Discussion by GABowers** [jspsych/jsPsych#3016] [General]
I've been developing an experiment using the HTML version of JSPsych (option 1 [here](https://www.jspsych.org/7.3/tutorials/hello-world/)).

I have a couple lab machines that are not connected to the internet.

Is it possible to download whatever resources necessary for the experiment HTML to run completely offline the first time on a new machine? If so, how? If not, what do I need to do (which method of using JSPsych) to achieve this?

**Comment by chguw** [comment]
follow this instruction:
https://www.jspsych.org/7.3/tutorials/hello-world/#option-2-download-and-host-jspsych

and you need to add
```
on_finish: function() {
                jsPsych.data.get().localSave("csv", "mydata.csv")
            }
```

in the `initJsPsych` method to save data.

**Reply by GABowers** [reply]
Just gave this a try and it worked. Thanks!
